### PR TITLE
Changes:

### PR DESCRIPTION
**Pull request by arturobernalg**
* Use StringBuilder
* Avoid unnecessary boxing
* Collapse identical catch
* Use comparator

### CODE REVIEW

**Comment by arturobernalg** [comment]
> > > sort_summary_message
> > 
> > 
> > Hi @TanguySoto You mean the branch and the commit? or just the commit? TY
> 
> Sorry, I meant the commit! Thanks.

H @TanguySoto 
Sorry answer so late.  Changed. Please check the commit and let me know if you're OK with the commit name.
TY

File: mission/simulator/opssat-gps-simulator/src/main/java/opssat/simulator/GPS.java [hunk 1]
```diff
@@ -101,9 +102,9 @@ public OrbitParameters getPosition(Date time) {
     public static BigDecimal truncateDecimal(double x, int numberofDecimals) {
         // From: http://stackoverflow.com/questions/7747469/how-can-i-truncate-a-double-to-only-two-decimal-places-in-java
         if (x > 0) {
-            return new BigDecimal(String.valueOf(x)).setScale(numberofDecimals, BigDecimal.ROUND_FLOOR);
+            return new BigDecimal(String.valueOf(x)).setScale(numberofDecimals, RoundingMode.FLOOR);
```
Comment (TanguySoto) [inline mission/simulator/opssat-gps-simulator/src/main/java/opssat/simulator/GPS.java:105]: Are we sure this is strictly equivalent? And what do we gain in doing this?

**Comment by TanguySoto** [comment]
> 
> 
> > > > sort_summary_message
> > > 
> > > 
> > > Hi @TanguySoto You mean the branch and the commit? or just the commit? TY
> > 
> > 
> > Sorry, I meant the commit! Thanks.
> 
> H @TanguySoto Sorry answer so late. Changed. Please check the commit and let me know if you're OK with the commit name. TY

I know it's difficult but if you could put a short summary after [refactor] that would be nice.

File: mission/simulator/opssat-gps-simulator/src/main/java/opssat/simulator/GPS.java (hunk 1 above)
Comment (arturobernalg) [reply mission/simulator/opssat-gps-simulator/src/main/java/opssat/simulator/GPS.java:105]: Yes. the method with BigDecimal it's deprecated.

` * @deprecated The method {@link #setScale(int, RoundingMode)} should
     * be used in preference to this legacy method.`

File: core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/archive/db/SortByField.java [hunk 2]
```diff
@@ -287,7 +287,7 @@ public int compare(Object in1, Object in2) {
         SortByField comparator = new SortByField(aClass, fieldString, ascending);
         // stack.sort(comparator);
         // Changed to be compatible with java 6:
```
[the remark this answers is not in the document]
Comment (dmarszk) [reply core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/archive/db/SortByField.java:289]: We are fine breaking compatibility with Java 6

**Pull request merged**